grimoire/core/generator/test.py

```diff
@@ -1,16 +1,48 @@
 import abc
+from abc import ABC
 
-from ..core.generator.module import Module
+from gdpc import Editor, Rect, WorldSlice
+
+from ...core.generator.module import Module
+
+
+def run_test(cls):
+    mod: TestModule = cls()
+    mod.run()
+    return cls
 
 
 class TestModule(Module):
+    catch_errors: bool = True
+
     @abc.abstractmethod
     def test(self):
         pass
 
     @Module.main
     def run(self):
-        try:
+        if self.catch_errors:
```

Comment:
We will be adding the `sys.flags` when rewriting the modules

Reply:
After a brief discussion we agree it makes more sense for `sys.flags`' debug flag to be used as a "strict mode" across all modules. Hiding errors of specific modules should not be encouraged as it may obfuscate the true nature of an error. When the debug flag is not set, any error the system can recover from should be logged as errors, but not raise exceptions (unless caught by a higher scope).